### PR TITLE
Added an e2e test timing HPA + CA scaling up from 1 to 8 pods and from 3 to >=4 clusters

### LINKED ISSUE #46847: Add an e2e test measuring the time it takes for HPA and CA to scale up after load increase
Currently the time it takes for HPA and CA to scale up after load increase is not measured at all, and this can lead to regressions that will be missed. This issue is about adding an e2e test that will prevent such regressions.

@wojtek-t @fgrzadkowski

### PR DESCRIPTION
**Pull request by wasylkowski**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #46847 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/autoscaling/autoscaling_timer.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
Comment (mwielgus) [inline test/e2e/autoscaling/autoscaling_timer.go]: Wrong year.

File: test/e2e/autoscaling/autoscaling_timer.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"time"
```
Comment (mwielgus) [inline test/e2e/autoscaling/autoscaling_timer.go]: organize imports. 
1. Golang
2. Kubernetes
3. Others.

File: test/e2e/autoscaling/autoscaling_timer.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+				Expect(schedulableCount).To(Equal(nodeGroupSize), "not all nodes are schedulable")
+			})
+
+			AfterEach(func() {
+				// Scale down back to only 3 nodes, as expected at the start of the test.
+				framework.ExpectNoError(framework.ResizeGroup(nodeGroupName, 3))
+				framework.ExpectNoError(framework.WaitForClusterSize(f.ClientSet, 3, 15*time.Minute))
+			})
+
+			Measure("takes less than 15 minutes", func(b Benchmarker) {
```
Comment (mwielgus) [inline test/e2e/autoscaling/autoscaling_timer.go]: You need to reduce the access to only these CI runs that target CA. Add `[Feature:ClusterSizeAutoscalingScaleUp]`.

File: test/e2e/autoscaling/autoscaling_timer.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+				// Make sure all nodes are schedulable, otherwise we are in some kind of a problem state.
+				nodes = framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+				schedulableCount := len(nodes.Items)
+				Expect(schedulableCount).To(Equal(nodeGroupSize), "not all nodes are schedulable")
+			})
+
+			AfterEach(func() {
+				// Scale down back to only 3 nodes, as expected at the start of the test.
+				framework.ExpectNoError(framework.ResizeGroup(nodeGroupName, 3))
```
Comment (mwielgus) [inline test/e2e/autoscaling/autoscaling_timer.go]: Please make 3 a constant.

File: test/e2e/autoscaling/autoscaling_timer.go (hunk 1 above)
Comment (wasylkowski) [inline test/e2e/autoscaling/autoscaling_timer.go]: Fixed

File: test/e2e/autoscaling/autoscaling_timer.go (hunk 2 above)
Comment (wasylkowski) [inline test/e2e/autoscaling/autoscaling_timer.go]: Done

File: test/e2e/autoscaling/autoscaling_timer.go (hunk 3 above)
Comment (wasylkowski) [inline test/e2e/autoscaling/autoscaling_timer.go]: Done

File: test/e2e/autoscaling/autoscaling_timer.go (hunk 4 above)
Comment (wasylkowski) [inline test/e2e/autoscaling/autoscaling_timer.go]: Done

**Comment by wasylkowski** [comment]
Marcin, ping...

**Comment by mwielgus** [comment]
@k8s-bot ok to test

**Pull request merged**